cmake/mode.DEBUG.cmake

```diff
@@ -1,6 +1,9 @@
 if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
   # -fasan -Og breaks some coroutines on aarch64, use -O0 instead
   set(default_Seastar_OptimizationLevel_DEBUG "0")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 16)
+  # workaround https://github.com/llvm/llvm-project/issues/62842
+  set(default_Seastar_OptimizationLevel_DEBUG "0")
```

Comment:
Please add a similar change to configure.py.

Let's also add a warning. Debug mode ends up so slow that nothing works.

Reply:
> Let's also add a warning. Debug mode ends up so slow that nothing works.

@avikivity You are wrong about this. See https://github.com/scylladb/scylladb/pull/14197#issuecomment-1596928038

Reply:
@avikivity Please read https://github.com/scylladb/scylladb/pull/14197#issuecomment-1596928038 and acknowledge that you've read it.

Reply:
I acknowledge that I read and understood that comment. It's entirely possible that an in-flight change caused additional trouble.

Can we quantify the slowdown? Perhaps ./test.py --mode debug's cpu (not wall) time before and after.

Reply:
CPU time of `./test.py --mode=debug`, tested on Ryzen 5950X:
With -Og: 630 minutes
With -O0: 1156 minutes
With -O3: same as with -Og

Reply:
#14456